source/contracts/reporting/Market.sol

```diff
@@ -70,26 +70,36 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
         require(feePerEthInAttoeth <= MAX_FEE_PER_ETH_IN_ATTOETH);
         require(_creator != NULL_ADDRESS);
         require(_cash.getTypeName() == "Cash");
+        require(_designatedReporterAddress != NULL_ADDRESS);
         reportingWindow = _reportingWindow;
         require(address(getForkingMarket()) == NULL_ADDRESS);
-        MarketFeeCalculator _marketFeeCaluclator = MarketFeeCalculator(controller.lookup("MarketFeeCalculator"));
-        reporterGasCostsFeeAttoeth = _marketFeeCaluclator.getTargetReporterGasCosts(_reportingWindow);
-        validityBondAttoeth = _marketFeeCaluclator.getValidityBond(_reportingWindow);
+        owner = _creator;
+        assessFees();
         endTime = _endTime;
         numOutcomes = _numOutcomes;
         numTicks = _numTicks;
         feePerEthInAttoeth = _feePerEthInAttoeth;
         marketCreationBlock = block.number;
         designatedReporterAddress = _designatedReporterAddress;
         cash = _cash;
-        owner = _creator;
         for (uint8 _outcome = 0; _outcome < numOutcomes; _outcome++) {
             shareTokens.push(createShareToken(_outcome));
         }
         approveSpenders();
         // If the value was not at least equal to these this will throw
         uint256 _refund = msg.value.sub(reporterGasCostsFeeAttoeth.add(validityBondAttoeth));
-        require(owner.call.value(_refund)());
+        if (_refund > 0) {
+            require(owner.call.value(_refund)());
+        }
+        return true;
+    }
+
+    function assessFees() private returns (bool) {
+        MarketFeeCalculator _marketFeeCaluclator = MarketFeeCalculator(controller.lookup("MarketFeeCalculator"));
+        IUniverse _universe = getUniverse();
+        reportingWindow.getReputationToken().trustedTransfer(owner, this, _marketFeeCaluclator.getDesignatedReportNoShowBond(_universe));
```

Comment:
Security: Someone can create a market and put another person as the market creator and this will withdraw tokens from `creator` without their authorization.  This combined with the initialization in the factory issue makes me wonder if we should come up with a better pattern for this.  As with many other pieces of code, the problem here is being able to provide transfer authorizations through an untrusted interface.

We have discussed previously using authorization signatures, but this requires an extract signing step by users which is a sucky experience.  We could make the whole stack trusted, and use `msg.sender` as the creator at the root and trust it to correctly set `creator` parameter that is passed through, but this increases the auditable code surface area and makes auditing generally harder.

Unfortunately, I don't have any particularly great ideas.  One simple (but still lame) option is to just have a separate market activation function that the user must call (directly) after the market is create (e.g., `market.activate()`.  This would be where it transfers the REP.  Another option would be to have the the REP get transferred through each contract (like msg.value gets passed along with each step).  This requires every step along the way can use `ReputationToken.trustedTransfer`, and will eat through a lot of gas constantly updating storage variables along the way (though I think there is an EIP out to make it so that only net state changes at end of transaction are billed).

Reply:
If we do end up having two-step construction for market, we should just do all fee assessment as step 2, and allow any one to pay the fees.  This would allow someone to have a setup where an automated system creates a bunch of markets, and then some other system goes through and funds them with ETH/REP.  This could allow for clever security systems in complex automated market creation services.

Reply:
A second call to activate the market and do fee assessment sounds like the nicest way to me. Will refactor to that!

Reply:
We decided after beginning to implement this and seeing how much of a sprawling hazard it actually is that we'll instead have one transaction still, but will no longer exist on `MarketCreation`, which will be removed. Instead `ReportingWindow.createMarket` will do a trusted transfer from the `msg.sender` to the `MarketFactory` which will then transfer the REP to the `Market` before initialization. If during initialization the `Market` has a REP balance < the bond it will fail